woocommerce/woocommerce
# CYS: Add homepage template AI completion and revamped header footer

### Changes proposed in this Pull Request:

Part of https://github.com/woocommerce/woocommerce/issues/40162.

This PR:

- Merged header and footer into homepage template
- Added homepage template AI completion
- Added metadata into homepage template to assist with AI completion



### How to test the changes in this Pull Request:

This feature is part of the larger customize your store development and behind a feature flag, hence QA review can be deferred until the call for testing before release.

Make sure you are testing this with a Jetpack connected instance of WooCommerce, as the AI text completion endpoint can only be called with a Jetpack connected account.

### Test large business template

1. Create a new WooCommerce installation with this version.
2. Make sure to enable `customize-store` feature flag
3. Go to `WooCommerce -> Home` and click `Customize Store` task
4. Click `Design with AI` button
5. In the store description, use a description that signifies being a large business, for example: `My store sells modern furniture, we have factories in over 20 countries.`
6. Open browser's developer console, networks tab
7. Filter for `text-completion`
8. Proceed all steps until the end
9. Look for a query response that has `homepage_template` in it
10. Observe that the selected template is either `template1`, `template2`, or `template3`

### Test small business template

1. Go back to store description, use a description that signifies a small medium business, for example: `My store sells locally grown, homemade pies to all counties in a small state.`
8. Proceed all steps until the end
9. Look for a query response that has `homepage_template` in it
10. Observe that the selected template is either `template4`, `template5`, or `template6`
12. Should be redirected to assembler hub screen
13. Examine that the theme style and template are set based on AI response.




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
